rfcs/2020-09-03-3226-api-topology.md

```diff
@@ -0,0 +1,141 @@
+# RFC #3226 - 2020-09-03 - API Topology
+
+This RFC discusses configuration, implementation and potential schema for the API to co-exist with, and observe, existing Vector topology.
+
+## Scope
+
+* Implementation of the API server alongside existing topology.
+* Configuration in `vector.toml` to set-up / enable the API.
+* GraphQL types to observe existing topology (pending #3648)
+
+## Motivation
+
+I implemented the first draft of a GraphQL API server in #3514, and made changes to the toplogy mod to accommodate it. @lukesteensen [commented](https://github.com/timberio/vector/pull/3514#discussion_r479517596) with a preference to keep the API out of the existing topology mod.
+
+This raises several design questions:
+
+* Where should the API live?
```

Comment:
I think we should start with it as a standalone, independent component. We can simply check the config for whether or not it's enabled and then spawn the server in `main`.